embeddings/src/lib.rs

```diff
@@ -38,97 +37,203 @@ fn handle_embeddings(req: Request) -> Result<Response> {
+
+    let conn = Connection::open_default()?;
+
+    for (i, e) in embeddings.iter_mut().enumerate() {
+        if let Some(embedding) = embedding_result.embeddings.get(i) {
+            e.embedding = Some(embedding.clone());
+
+            let vec = json!(e.embedding);
+            let blob = serde_json::to_vec(&vec)?;
+
+            let query_params = [
+                sqlite::ValueParam::Text(e.reference.as_deref().unwrap_or_default()),
+                sqlite::ValueParam::Text(e.text.as_str()),
+                sqlite::ValueParam::Blob(blob.as_slice()),
+            ];
+
+            let result = conn.execute(
+                "INSERT INTO embeddings ('reference', 'text', 'embedding') VALUES (?, ?, ?);",
+                &query_params,
+            );
+
+            trace!("Result: {:?}", result);
+        }
+    }
+    // Some error handling needed...
+    Ok(embeddings)
 }
 
 impl<'a> TryFrom<sqlite::Row<'a>> for Embedding {
     type Error = anyhow::Error;
 
     fn try_from(row: sqlite::Row<'a>) -> std::result::Result<Self, Self::Error> {
-        info!("DB Row: {:?}", row.get::<&str>("embedding"));
-        info!("DB Row: {:?}", row.get::<u32>("embedding"));
-        let id = row.get::<u32>("id").unwrap();
+        // TODO: Fix unwraps()
+        let id = Some(row.get::<u32>("id").unwrap());
         let reference = row
             .get::<&str>("reference")
             .context("reference column is empty")?;
         let text = row.get::<&str>("text").context("text column is empty")?;
-        let embedding = row.get::<&str>("embedding").unwrap_or_default();
-        info!("Embedding from DB {:?}", embedding);
-        //let embedding: Option<Vec<f32>> = embedding.unwrap()
-        //let embedding: Vec<f32> = bincode::deserialize_from(Cursor::new(embedding))?;
+        let embedding: Vec<f32> = match row.get::<&ValueResult>("embedding").unwrap() {
+            ValueResult::Blob(b) => serde_json::from_value(serde_json::from_slice(b.as_slice()).unwrap())?,
+            _ => todo!(),
+        };
         Ok(Self {
             id,
-            reference: reference.to_owned(),
+            reference: Some(reference.to_owned()),
             text: text.to_owned(),
-            embedding: None,
+            embedding: Some(embedding),
         })
     }
 }
+
+fn get_similar(query: Query) -> Result<SimilarityResultSet> {
+    let text: Vec<&str> = vec![query.text.as_str()];
+    let embedding = generate_embeddings(AllMiniLmL6V2, &text[..]);
+    let sql_query = "SELECT * FROM embeddings";
+    let conn = Connection::open_default()?;
+    let embedding_records: Vec<Embedding> = conn
+        .execute(sql_query, &[])?
+        .rows()
+        .map(|row| -> anyhow::Result<Embedding> { row.try_into() })
+        .collect::<anyhow::Result<Vec<Embedding>>>()?;
+
+    let mut result_set = SimilarityResultSet {
+        text: query.text.to_string(),
+        results: Vec::new(),
+    };
+
+    for e in embedding_records.iter() {
```

Comment:
You can get rid of some clones below with:

```suggestion
    for e in embedding_records.into_iter() {
```